### PR TITLE
Add fishing event factories

### PR DESCRIPTION
**Pull request by Aaron1011**
This adds factories for creating subclasses of `PlayerFishEvent`

### CODE REVIEW

File: src/main/java/org/spongepowered/api/event/SpongeEventFactory.java [hunk 1]
```diff
@@ -831,6 +835,72 @@ public static PlayerBreakBlockEvent createPlayerBreakBlock(Game game, Cause caus
...
+     * Creates a new {@link PlayerRetractFishingLineEvent}.
+     *
+     * @param game The game instance for this {@link GameEvent}
+     * @param player The player involved in this event
+     * @param fishHook The {@link FishHook} affected by this event
+     * @param caughtItem The {@link ItemStack} caught by the player, can be null
+     * @param caughtEntity The {@link Entity} caught by the player, can be null
+     * @return A new instance of the event
+     */
+    public static PlayerRetractFishingLineEvent createPlayerRetractFishingLineEvent (Game game, Player player, FishHook fishHook, ItemStack caughtItem, Entity caughtEntity) {
```
Comment (gratimax) [inline src/main/java/org/spongepowered/api/event/SpongeEventFactory.java]: `createPlayerRetractFishingLineEvent (`

**Comment by kenzierocks** [comment]
Quite a few extra newlines.

File: src/main/java/org/spongepowered/api/event/SpongeEventFactory.java [hunk 2]
```diff
@@ -831,6 +835,72 @@ public static PlayerBreakBlockEvent createPlayerBreakBlock(Game game, Cause caus
...
+        values.put("entity", player);
+        values.put("player", player);
+        values.put("human", player);
+        values.put("living", player);
+        values.put("fishHook", fishHook);
+        values.put("caughtEntity", caughtEntity);
+        return createEvent(PlayerRetractFishingLineEvent.class, values);
+    }
+
+
```
Comment (gratimax) [inline src/main/java/org/spongepowered/api/event/SpongeEventFactory.java]: newline

**Comment by gratimax** [comment]
Aside from some formatting fixes, this PR is perfectly fine. You have the ability to push to master @Aaron1011, and for stuff like this I recommend doing it. :smile:

**Pull request merged**